sema/commons/j2/j2_functions.py

```
@@ -46,43 +55,78 @@ def xsd_format_boolean(content, quote, suffix):
+        year = content.year  # extract year from date
+    else:  # other input types handled
+        content = str(content).strip()  # via trimmed string
+        year = int(content)  # converted to int
+    # we should be sure of int now
+    # see https://www.datypic.com/sc/xsd11/t-xsd_gYear.html
+    # for examples of correct value formatting
+    content = f"{"-" if year < 0 else ""}{abs(year):04d}"
+    return xsd_value(content, quote, "xsd:gYear")
+
+
+def xsd_format_gmonthyear(content: Any, quote: str, *_: Any) -> str:
```

Ruff (0.8.2)

Dynamically typed expressions (typing.Any) are disallowed in `content`

(ANN401)